pandas/core/dtypes/cast.py

```diff
@@ -1190,14 +1207,10 @@ def astype_nansafe(
         flags = arr.flags
         flat = arr.ravel("K")
         result = astype_nansafe(flat, dtype, copy=copy, skipna=skipna)
-        order = "F" if flags.f_contiguous else "C"
+        order: Literal["C", "F"] = "F" if flags.f_contiguous else "C"
         # error: Item "ExtensionArray" of "Union[ExtensionArray, ndarray]" has no
         # attribute "reshape"
```

Comment:
@jbrockmendel this is not necessarily a false positive. I think this maybe assuming that if ndim > 1, we have a numpy array. but a 2d EA will have additional methods such as reshape.

Reply:
yah even if we got here with NDArrayBackedExtensionArray (which does have reshape) itd break bc it doesn't have `flags`.  ignoring for the time being seems reasonable